src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -49,6 +60,57 @@ internal bool AutomaticRedirection
                 _automaticRedirection = value;
             }
         }
+        internal bool UseProxy
+        {
+            get
+            {
+                if (_proxyPolicy == CurlProxyUsePolicy.UseDefaultProxy)
+                {
+                    return false;
+                }
+                return _proxyPolicy != CurlProxyUsePolicy.DoNotUseProxy;
```

Comment:
About "how this relates at all to an environment variable":
libcurl relies on the value of environment variable "http_proxy". Even if we do not set any proxy, if the environment variable is set to some thing, libcurl uses it. Otherwise, left to itself, it does not use any proxy.

If the user has not explicitly set Proxy value, whether the next http call goes via a proxy or not is decided by such an environment variable. 
For instace, what happens in the following scenario (assuming no environment variable is set)? It is not a valid code, by any chance?:

         if (!handler.UseProxy) {
               handler.Proxy = someProxyInstance;
         }